docs/language/informal/toplevel-inference.md

```diff
@@ -0,0 +1,338 @@
+﻿# Toplevel inference
+
+Owner: leafp@google.com
+
+Status: Proposal
+
+
+## Top level inference overview
+
+Top level inference has two phases:
+
+1. **Method override inference**
+    * If you omit a return type or parameter type from an overridden method,
```

Comment:
Can you be more concrete about cases where the interpretation differs?  Note I am not defining how inference behaves on erroneous cases such as defining a method with the same name as a field from a superclass, or defining a method with the same name as a superclass method with an incompatible signature.  Those are errors: the tool can choose how best to handle surfacing those errors in a way that doesn't cause noise from cascading inference errors.